### PR TITLE
33 comment react component

### PR DESCRIPTION
**Pull request by hrytskivr**
## TRELLO

* [Main Trello ticket](https://trello.com/c/euG3w47H/33-comment-react-component)

## Code reviewers

- [ ] @Ashiato
- [ ] @black-art
- [x] @hrytskivr
- [ ] @KarnaushenkoVO
- [ ] @PanLeon
- [ ] @spycheast
- [ ] @yuriiZherebukh

### Second Level Review

- [ ] @byshovets
- [ ] @lhalam

## Summary of issue

Comment component [REACT]

## Summary of change

Add, Edit and Delete comment now works like a charm!

## Testing approach

ToDo

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=90%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

File: myTrip/static/src/app.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ import injectTapEventPlugin from 'react-tap-event-plugin';
 import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
 
 import Home from "./home/home.js";
-import Comment from "./comment/comment.js";
+import Comments from "./comment/Comments.js";
```
Comment (hrytskivr) [inline myTrip/static/src/app.js:12]: Looks better like that :)

File: myTrip/static/src/comment/CommentForm.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    handleTouchTap = (event) => {
+        this.setState({open: true});
+        const message = this.state.comment_text;
+        axios.post('api/v1/trip/2/comment/', {
+            message
+        })
+            .then(() => {
+                 this.props.getData();
```
Comment (hrytskivr) [inline myTrip/static/src/comment/CommentForm.js]: Rerender

File: myTrip/static/src/comment/CommentItem.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    };
+
+    handleOpenDelete = () => {
+      this.setState({openDelete: true});
+    };
+
+    handleCloseEdit = () => {
+      this.setState({openEdit: false});
+      this.setState({'disabledEdit': true});
+      this.setState({'edit_comment_text': ''});
```
Comment (hrytskivr) [inline myTrip/static/src/comment/CommentItem.js:44]: Deleting text in form after closing

File: myTrip/static/src/comment/CommentItem.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+            label="Cancel"
+            secondary={true}
+            onTouchTap={this.handleCloseEdit}
+          />,
+          <FlatButton
+            label="Submit"
+            disabled={this.state.disabledEdit}
+            onTouchTap={this.handleSubmit}
+          />,
+        ];
```
Comment (hrytskivr) [inline myTrip/static/src/comment/CommentItem.js:87]: Buttons for Edit dialog

File: myTrip/static/src/comment/CommentItem.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+            label="Cancel"
+            secondary={true}
+            onTouchTap={this.handleCloseDelete}
+          />,
+          <FlatButton
+            label="Delete"
+            disabled={this.state.disabledDelete}
+            onTouchTap={this.handleDelete}
+          />,
+        ];
```
Comment (hrytskivr) [inline myTrip/static/src/comment/CommentItem.js:100]: Buttons for Delete dialog

File: myTrip/static/src/comment/Comments.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            .then(() => this.getData());
+    }
+
+    getData = () => {
+        axios.get('api/v1/trip/2/comment/').then(response => {
+            const comments = response.data;
+            this.setState({comments});
+        });
+    }
+    componentDidMount() {
```
Comment (hrytskivr) [inline myTrip/static/src/comment/Comments.js]: Going to fix it right now

File: package.json [hunk 7]
```diff
@@ -21,10 +21,10 @@
     "babel-core": "^6.25.0",
     "babel-plugin-transform-class-properties": "^6.24.1",
     "css-loader": "^0.28.4",
+    "dev": "^0.1.3",
```
Comment (hrytskivr) [inline package.json]: Deleting this now

File: package.json [hunk 8]
```diff
@@ -21,10 +21,10 @@
...
     "extract-text-webpack-plugin": "^2.1.2",
-    "less": "^2.7.2",
-    "less-loader": "^4.0.5",
     "material-ui": "^0.18.7",
+    "moment": "^2.18.1",
```
Comment (hrytskivr) [inline package.json:26]: For datetime format

**Review by hrytskivr — COMMENTED** [review]
Everything seems fine to me

**Review by lhalam — APPROVED** [review]

**Pull request merged**